blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxIO.java

```diff
@@ -113,34 +106,58 @@ private static PoolingOptions getPoolingOptions(int numHosts){
-                long totalTrashedConnections = 0;
-                long totalMaxLoad =0;
-                for (Host host : hosts) {
-                    int openConnections = state.getOpenConnections(host);
-                    int inFlightQueries = state.getInFlightQueries(host);
-                    int trashedConnections = state.getTrashedConnections(host);
-                    int maxLoad = openConnections * 128;
-                    totalOpenConnections += openConnections;
-                    totalInFlightQueries += inFlightQueries;
-                    totalTrashedConnections += trashedConnections;
-                    totalMaxLoad += maxLoad;
-                }
-
-//                hostMeter.mark(totalHosts);
-//                openConnectionsMeter.mark(totalOpenConnections);
-//                inFlightQueriesMeter.mark(totalInFlightQueries);
-//                trashedConnectionsMeter.mark(totalTrashedConnections);
-//                maxLoadMeter.mark(totalMaxLoad);
+        final MetricRegistry bfMetricsRegistry = com.rackspacecloud.blueflood.utils.Metrics.getRegistry();
+        cluster.getMetrics().getRegistry().addListener(new com.codahale.metrics.MetricRegistryListener() {
+            @Override
+            public void onGaugeAdded(String name, Gauge<?> gauge) {
+                bfMetricsRegistry.register(metricsPrefix + name, gauge);
+            }
+
+            @Override
+            public void onGaugeRemoved(String name) {
+                bfMetricsRegistry.remove(metricsPrefix + name);
+            }
+
+            @Override
+            public void onCounterAdded(String name, Counter counter) {
+                bfMetricsRegistry.register(metricsPrefix + name, counter);
+            }
+
+            @Override
+            public void onCounterRemoved(String name) {
+                bfMetricsRegistry.remove(metricsPrefix + name);
+            }
+
+            @Override
+            public void onHistogramAdded(String name, Histogram histogram) {
+                bfMetricsRegistry.register(metricsPrefix + name, histogram);
+            }
+
+            @Override
+            public void onHistogramRemoved(String name) {
+                bfMetricsRegistry.remove(metricsPrefix + name);
+            }
+
+            @Override
+            public void onMeterAdded(String name, Meter meter) {
+                bfMetricsRegistry.register(metricsPrefix + name, meter);
+            }
+
+            @Override
+            public void onMeterRemoved(String name) {
+                bfMetricsRegistry.remove(metricsPrefix + name);
+            }
+
+            @Override
+            public void onTimerAdded(String name, Timer timer) {
+                bfMetricsRegistry.register(metricsPrefix + name, timer);
+            }
+
+            @Override
+            public void onTimerRemoved(String name) {
+                bfMetricsRegistry.remove(metricsPrefix + name);
             }
-        }, 1, 1, TimeUnit.MINUTES);
+        });
```

Comment:
I am having hard time understand the relation between code being removed and the new code added. Can you explain it a little bit?